src/components/modes/teacher/Settings.js

```diff
@@ -113,9 +113,17 @@ class Settings extends Component {
     this.saveSettings(settingsToChange);
   };
 
-  handleChangeTimeControlVisibility = () => {
-    const { timeControlsVisible } = this.state;
-    this.setState({ timeControlsVisible: !timeControlsVisible });
+  handleChangeTimerVisibility = () => {
+    const { timerVisible } = this.state;
+    this.setState({ timerVisible: !timerVisible }, () => {
+      if (timerVisible === false) {
```

Comment:
Are you sure `timerVisible` has already changed to `false`? You don't have to get it again from state?

Reply:
Thank you @juancarlosfarah. I was getting `timerVisible` from the local state earlier. Fixed now.